src/rating/Rating.js

```diff
@@ -138,25 +153,57 @@ export default class Rating extends Component {
 
   getCurrentRating() {
     const { value } = this.state;
-    const { imageSize, ratingCount } = this.props;
+    const { fractions, imageSize, ratingCount } = this.props;
     const startingValue = ratingCount / 2;
     let currentRating = 0;
 
     if (value > ratingCount * imageSize / 2) {
       currentRating = ratingCount;
     } else if (value > imageSize) {
-      currentRating = Math.ceil(startingValue + value / imageSize);
+      currentRating = startingValue + value / imageSize;
+      currentRating = !fractions
+        ? Math.ceil(currentRating)
+        : +currentRating.toFixed(fractions);
     } else if (value < -ratingCount * imageSize / 2) {
       currentRating = 0;
     } else if (value < imageSize) {
-      currentRating = Math.ceil(startingValue + value / imageSize);
+      currentRating = startingValue + value / imageSize;
+      currentRating = !fractions
+        ? Math.ceil(currentRating)
+        : +currentRating.toFixed(fractions);
     } else {
-      currentRating = Math.ceil(startingValue);
+      currentRating = !fractions
+        ? Math.ceil(startingValue)
+        : +startingValue.toFixed(fractions);
     }
 
     return currentRating;
   }
 
+  setCurrentRating(rating) {
+    const { imageSize, ratingCount } = this.props;
+    // `initialRating` corresponds to `startingValue` in the getter. Naming it
+    // differently here avoids confusion with `value` below.
+    const initialRating = ratingCount / 2;
+    let value = -ratingCount * imageSize / 2;
```

Comment:
the value assigned to the value variable is not being used. Can we refactor to `let value = nil`